community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/LuceneBatchInserterIndex.java

```diff
@@ -471,7 +482,17 @@ private File getStoreDir( File dbStoreDir )
     @Override
     public void flush()
     {
-        writerModified = true;
+        try
+        {
+            while ( !searcherManager.maybeRefresh() )
+            {
+                LockSupport.parkNanos( MILLISECONDS.toNanos( 5 ) );
```

Comment:
You should wait at least 10 milliseconds, as lower values will cause a system-wide change of the scheduling quantum on Windows.